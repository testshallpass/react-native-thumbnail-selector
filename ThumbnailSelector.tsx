import React from 'react';
import {
  Text,
  FlatList,
  Image,
  TouchableOpacity,
  Animated,
  useWindowDimensions,
  ImageProps,
  ImageSourcePropType,
  TextProps,
  TouchableOpacityProps,
  ViewStyle,
  LayoutChangeEvent,
} from 'react-native';

export type ThumbnailItem = {
  caption?: string;
  imageSrc: ImageSourcePropType;
};

export type ThumbnailItemIndex = {
  item: ThumbnailItem;
  index: number;
};

export type ThumbnailSelectorProps = {
  thumbnails: ThumbnailItem[];
  toggle?: (func: () => Promise<Animated.EndResult>) => void;
  renderThumbnail?: (
    item: ThumbnailItem,
    index: number,
    onSelect?: (item: ThumbnailItem, index: number) => void,
  ) => JSX.Element;
  onSelect?: (item: ThumbnailItem, index: number) => void;
  initialIndex?: number;
  captionCharacterMaxLimit?: number;
  captionEllipsis?: string;
  activeColor?: string;
  inactiveColor?: string;
  imageProps?: ImageProps;
  textProps?: TextProps;
  touchableOpacityProps?: TouchableOpacityProps;
  animatedViewStyle?: ViewStyle;
  animationConfig?: Animated.SpringAnimationConfig;
  flatListViewStyle?: ViewStyle;
  animatedViewTestID?: string;
};

const ThumbnailSelector: React.FunctionComponent<ThumbnailSelectorProps> = ({
  thumbnails,
  toggle = undefined,
  renderThumbnail = undefined,
  onSelect = undefined,
  initialIndex = -1,
  captionCharacterMaxLimit = 15,
  captionEllipsis = '...',
  activeColor = 'white',
  inactiveColor = 'black',
  imageProps = {
    style: { width: 136, height: 136, borderWidth: 1 },
  },
  textProps = {
    style: { fontSize: 16, textAlign: 'center', fontWeight: 'bold' },
    numberOfLines: 1,
  },
  touchableOpacityProps = {
    style: { padding: 8 },
  },
  animatedViewStyle = {
    elevation: 1,
    zIndex: 1,
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
  },
  animationConfig = {
    toValue: 0,
    friction: 9,
    useNativeDriver: false,
  },
  flatListViewStyle = { backgroundColor: 'grey', padding: 8 },
  animatedViewTestID = 'ThumbnailSelector',
}) => {
  const window = useWindowDimensions();
  const [itemIndex, setItemIndex] = React.useState(initialIndex);
  const [animViewHeight, setAnimViewHeight] = React.useState(0);
  const animatedValue = React.useRef(new Animated.Value(0));
  const toValue = React.useRef(animationConfig.toValue);

  function _toggle(): Promise<Animated.EndResult> {
    if (toValue.current) {
      animationConfig.toValue = 0;
    } else {
      animationConfig.toValue = 1;
    }
    toValue.current = animationConfig.toValue;
    return new Promise<Animated.EndResult>(resolve => {
      Animated.spring(animatedValue.current, animationConfig).start(resolve);
    });
  }

  if (toggle) {
    toggle(_toggle);
  }

  function _renderItem(obj: ThumbnailItemIndex): React.JSX.Element {
    const { item, index } = obj;
    if (renderThumbnail) {
      return renderThumbnail(item, index, onSelect);
    }
    const selected = itemIndex === index;
    const color = selected ? activeColor : inactiveColor;
    let caption = item.caption;
    if (caption) {
      if (caption.length > captionCharacterMaxLimit) {
        const end = captionCharacterMaxLimit - captionEllipsis.length;
        caption = `${caption.substring(0, end)}${captionEllipsis}`;
      }
    }
    return (
      <TouchableOpacity
        {...touchableOpacityProps}
        onPress={() => {
          setItemIndex(index);
          if (onSelect) {
            onSelect(item, index);
          }
        }}
      >
        <Image
          {...imageProps}
          style={[{ borderColor: color }, imageProps.style]}
          source={item.imageSrc}
        />
        {caption && (
          <Text {...textProps} style={[{ color }, textProps.style]}>
            {caption}
          </Text>
        )}
      </TouchableOpacity>
    );
  }

  function _onLayout(event: LayoutChangeEvent): void {
    const { height } = event.nativeEvent.layout;
    if (animViewHeight !== height) {
      setAnimViewHeight(height);
    }
  }

  function _getAnimViewStyle(): ViewStyle[] {
    const start = window.height;
    const end = window.height - animViewHeight;
    return [
      animatedViewStyle,
      {
        transform: [
          {
            translateY: animatedValue.current.interpolate({
              inputRange: [0, 1],
              outputRange: [start, end],
            }),
          },
        ],
      },
    ];
  }

  return (
    <Animated.View
      testID={animatedViewTestID}
      style={_getAnimViewStyle()}
      onLayout={_onLayout}
    >
      <FlatList
        style={flatListViewStyle}
        data={thumbnails}
        initialNumToRender={thumbnails.length}
        extraData={itemIndex}
        horizontal={true}
        renderItem={_renderItem}
        keyExtractor={(_item, index) => `${index}`}
      />
    </Animated.View>
  );
};

export default ThumbnailSelector;
