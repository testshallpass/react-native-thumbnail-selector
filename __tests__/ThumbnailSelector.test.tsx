import React, {act} from 'react';
import {Text} from 'react-native';
import {
  render,
  fireEvent,
  screen,
  cleanup,
} from '@testing-library/react-native';
import ThumbnailSelector, {ThumbnailItem} from '../ThumbnailSelector';

describe('ThumbnailSelector', () => {
  afterEach(cleanup);

  function _truncate(
    value: string,
    maxLength: number = 15,
    ellipsis: string = '...',
  ): string {
    if (value.length > maxLength) {
      const end = maxLength - ellipsis.length;
      return `${value.substring(0, end)}${ellipsis}`;
    }
    return value;
  }

  const thumbnails = [
    {
      caption: 'react-native',
      imageSrc: {uri: 'https://reactnative.dev/img/tiny_logo.png'},
    },
    {
      caption: 'New York City',
      imageSrc: {uri: 'https://reactnative.dev/img/tiny_logo.png'},
    },
    {
      caption:
        'Elit cupidatat qui ea deserunt reprehenderit sit velit eu aliqua incididunt sit elit reprehenderit.',
      imageSrc: {uri: 'https://reactnative.dev/img/tiny_logo.png'},
    },
    {
      imageSrc: {uri: 'https://reactnative.dev/img/tiny_logo.png'},
    },
  ];

  test('it renders', () => {
    const component = render(<ThumbnailSelector thumbnails={thumbnails} />);
    thumbnails.forEach(thumbnail => {
      if (thumbnail.caption) {
        const item = screen.getByText(_truncate(thumbnail.caption));
        expect(item).toBeDefined();
        fireEvent.press(item);
      }
    });
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('onSelect prop', async () => {
    let toggle = () => {};
    let onSelect = jest.fn();
    const component = render(
      <ThumbnailSelector
        thumbnails={thumbnails}
        toggle={func => (toggle = func)}
        onSelect={onSelect}
      />,
    );
    await act(async () => toggle());
    thumbnails.forEach((thumbnail, index) => {
      if (thumbnail.caption) {
        const item = screen.getByText(_truncate(thumbnail.caption));
        expect(item).toBeDefined();
        fireEvent.press(item);
        expect(onSelect).toHaveBeenCalledTimes(index + 1);
      }
    });
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('renderThumbnail prop', async () => {
    let toggle = () => {};
    const renderThumbnail = (item: ThumbnailItem, index: number) => {
      return <Text testID={`${index}`}>{item.caption}</Text>;
    };
    render(
      <ThumbnailSelector
        thumbnails={thumbnails}
        toggle={func => (toggle = func)}
        renderThumbnail={renderThumbnail}
      />,
    );
    await act(async () => toggle());
    thumbnails.forEach((thumbnail, index) => {
      if (thumbnail.caption) {
        const itemText = screen.getByText(thumbnail.caption);
        expect(itemText).toBeDefined();
      }
      const itemTestId = screen.getByTestId(`${index}`);
      expect(itemTestId).toBeDefined();
    });
    await act(async () => toggle());
  });

  test('onLayout', () => {
    render(<ThumbnailSelector thumbnails={thumbnails} />);

    const selector = screen.getByTestId('ThumbnailSelector');
    expect(selector).toBeDefined();

    fireEvent(selector, 'layout', {
      nativeEvent: {layout: {height: 0}},
    });

    fireEvent(selector, 'layout', {
      nativeEvent: {layout: {height: 100}},
    });
  });

  test('no thumbnails', () => {
    const component = render(<ThumbnailSelector thumbnails={[]} />);
    thumbnails.forEach(thumbnail => {
      if (thumbnail.caption) {
        const text = _truncate(thumbnail.caption);
        try {
          screen.getByText(text);
        } catch (error: unknown) {
          expect(error).toBeDefined();
          if (error instanceof Error) {
            expect(error).toBeInstanceOf(Error);
            const partialErrorMessage = `Unable to find an element with text: ${text}`;
            expect(error.message).toContain(partialErrorMessage);
          }
        }
      }
    });
    expect(component.toJSON()).toMatchSnapshot();
  });
});
