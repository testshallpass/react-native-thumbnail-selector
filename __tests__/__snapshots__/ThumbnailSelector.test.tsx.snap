// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThumbnailSelector it renders 1`] = `
<View
  collapsable={false}
  onLayout={[Function]}
  style={
    {
      "elevation": 1,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "transform": [
        {
          "translateY": 1334,
        },
      ],
      "zIndex": 1,
    }
  }
  testID="ThumbnailSelector"
>
  <RCTScrollView
    data={
      [
        {
          "caption": "react-native",
          "imageSrc": {
            "uri": "https://reactnative.dev/img/tiny_logo.png",
          },
        },
        {
          "caption": "New York City",
          "imageSrc": {
            "uri": "https://reactnative.dev/img/tiny_logo.png",
          },
        },
        {
          "caption": "Elit cupidatat qui ea deserunt reprehenderit sit velit eu aliqua incididunt sit elit reprehenderit.",
          "imageSrc": {
            "uri": "https://reactnative.dev/img/tiny_logo.png",
          },
        },
        {
          "imageSrc": {
            "uri": "https://reactnative.dev/img/tiny_logo.png",
          },
        },
      ]
    }
    extraData={2}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    initialNumToRender={4}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    style={
      {
        "backgroundColor": "grey",
        "padding": 8,
      }
    }
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={
          [
            {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
              "padding": 8,
            }
          }
        >
          <Image
            source={
              {
                "uri": "https://reactnative.dev/img/tiny_logo.png",
              }
            }
            style={
              [
                {
                  "borderColor": "black",
                },
                {
                  "borderWidth": 1,
                  "height": 136,
                  "width": 136,
                },
              ]
            }
          />
          <Text
            numberOfLines={1}
            style={
              [
                {
                  "color": "black",
                },
                {
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
              ]
            }
          >
            react-native
          </Text>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={
          [
            {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
              "padding": 8,
            }
          }
        >
          <Image
            source={
              {
                "uri": "https://reactnative.dev/img/tiny_logo.png",
              }
            }
            style={
              [
                {
                  "borderColor": "black",
                },
                {
                  "borderWidth": 1,
                  "height": 136,
                  "width": 136,
                },
              ]
            }
          />
          <Text
            numberOfLines={1}
            style={
              [
                {
                  "color": "black",
                },
                {
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
              ]
            }
          >
            New York City
          </Text>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={
          [
            {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
              "padding": 8,
            }
          }
        >
          <Image
            source={
              {
                "uri": "https://reactnative.dev/img/tiny_logo.png",
              }
            }
            style={
              [
                {
                  "borderColor": "white",
                },
                {
                  "borderWidth": 1,
                  "height": 136,
                  "width": 136,
                },
              ]
            }
          />
          <Text
            numberOfLines={1}
            style={
              [
                {
                  "color": "white",
                },
                {
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
              ]
            }
          >
            Elit cupidat...
          </Text>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={
          [
            {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
              "padding": 8,
            }
          }
        >
          <Image
            source={
              {
                "uri": "https://reactnative.dev/img/tiny_logo.png",
              }
            }
            style={
              [
                {
                  "borderColor": "black",
                },
                {
                  "borderWidth": 1,
                  "height": 136,
                  "width": 136,
                },
              ]
            }
          />
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`ThumbnailSelector no thumbnails 1`] = `
<View
  collapsable={false}
  onLayout={[Function]}
  style={
    {
      "elevation": 1,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "transform": [
        {
          "translateY": 1334,
        },
      ],
      "zIndex": 1,
    }
  }
  testID="ThumbnailSelector"
>
  <RCTScrollView
    data={[]}
    extraData={-1}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    initialNumToRender={0}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    style={
      {
        "backgroundColor": "grey",
        "padding": 8,
      }
    }
    viewabilityConfigCallbackPairs={[]}
  >
    <View />
  </RCTScrollView>
</View>
`;

exports[`ThumbnailSelector onSelect prop 1`] = `
<View
  collapsable={false}
  onLayout={[Function]}
  style={
    {
      "elevation": 1,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "transform": [
        {
          "translateY": 1334,
        },
      ],
      "zIndex": 1,
    }
  }
  testID="ThumbnailSelector"
>
  <RCTScrollView
    data={
      [
        {
          "caption": "react-native",
          "imageSrc": {
            "uri": "https://reactnative.dev/img/tiny_logo.png",
          },
        },
        {
          "caption": "New York City",
          "imageSrc": {
            "uri": "https://reactnative.dev/img/tiny_logo.png",
          },
        },
        {
          "caption": "Elit cupidatat qui ea deserunt reprehenderit sit velit eu aliqua incididunt sit elit reprehenderit.",
          "imageSrc": {
            "uri": "https://reactnative.dev/img/tiny_logo.png",
          },
        },
        {
          "imageSrc": {
            "uri": "https://reactnative.dev/img/tiny_logo.png",
          },
        },
      ]
    }
    extraData={2}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    initialNumToRender={4}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    style={
      {
        "backgroundColor": "grey",
        "padding": 8,
      }
    }
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={
          [
            {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
              "padding": 8,
            }
          }
        >
          <Image
            source={
              {
                "uri": "https://reactnative.dev/img/tiny_logo.png",
              }
            }
            style={
              [
                {
                  "borderColor": "black",
                },
                {
                  "borderWidth": 1,
                  "height": 136,
                  "width": 136,
                },
              ]
            }
          />
          <Text
            numberOfLines={1}
            style={
              [
                {
                  "color": "black",
                },
                {
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
              ]
            }
          >
            react-native
          </Text>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={
          [
            {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
              "padding": 8,
            }
          }
        >
          <Image
            source={
              {
                "uri": "https://reactnative.dev/img/tiny_logo.png",
              }
            }
            style={
              [
                {
                  "borderColor": "black",
                },
                {
                  "borderWidth": 1,
                  "height": 136,
                  "width": 136,
                },
              ]
            }
          />
          <Text
            numberOfLines={1}
            style={
              [
                {
                  "color": "black",
                },
                {
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
              ]
            }
          >
            New York City
          </Text>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={
          [
            {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
              "padding": 8,
            }
          }
        >
          <Image
            source={
              {
                "uri": "https://reactnative.dev/img/tiny_logo.png",
              }
            }
            style={
              [
                {
                  "borderColor": "white",
                },
                {
                  "borderWidth": 1,
                  "height": 136,
                  "width": 136,
                },
              ]
            }
          />
          <Text
            numberOfLines={1}
            style={
              [
                {
                  "color": "white",
                },
                {
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
              ]
            }
          >
            Elit cupidat...
          </Text>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={
          [
            {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
              "padding": 8,
            }
          }
        >
          <Image
            source={
              {
                "uri": "https://reactnative.dev/img/tiny_logo.png",
              }
            }
            style={
              [
                {
                  "borderColor": "black",
                },
                {
                  "borderWidth": 1,
                  "height": 136,
                  "width": 136,
                },
              ]
            }
          />
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
